### PR TITLE
Added printing of annotations to smt lib printers

### PR DESCRIPTION
**Pull request by agirardi-fbk**
Added support of SMT-LIB annotations printing

### CODE REVIEW

File: pysmt/smtlib/printers.py [hunk 1]
```diff
@@ -23,14 +23,45 @@
...
+        if self.annotations is not None and formula in self.annotations:
+            res = ["(! "]
+            res.append(f(self, formula, *args, **kwargs))
+            for key, values in self.annotations[formula].items():
+                res.append(f" :{key}")
+                for value in values:
+                    res.append(f" {value}")
+            res.append(")")
+            return ''.join(res)
+        return f(self, formula, *args, **kwargs)
```
Comment (johnyf) [inline pysmt/smtlib/printers.py:54]: A comment on nesting and indentation: by complementing the predicate of the `if` statement, eight lines can be moved to a smaller indentation level, and the maximal nesting depth of syntactic constructs reduced from 5 to 4. Together with other changes:

```python
def write_annotations_dag(f):
    def resf(self, formula, *arg, **kwargs):
        annots = self.annotations
        app = f(self, formula, *args, **kwargs)
        if annots is None or formula not in annots:
            return app
        items = list()
        for key, values in annots[formula].items():
            items.append(f' :{key}')
            items.extend(f' {v}' for v in values)
        kv = ''.join(items)
        return f'(! {app}{kv})'
```

Comment (mikand) [reply pysmt/smtlib/printers.py:54]: Thanks for the contribution @johnyf!
@agirardi-fbk can you have a look and update the code / comment here?

Comment (agirardi-fbk) [reply pysmt/smtlib/printers.py]: Thank you for your suggestion @johnyf, I have updated the pull request

Comment (johnyf) [reply pysmt/smtlib/printers.py]: Thank you all for reviewing these changes, and updating the pull request.

**Review by mikand — APPROVED** [review]

**Pull request merged**